tamr_toolbox/low_latency/llm.py

```diff
@@ -0,0 +1,208 @@
+from typing import List, Dict, Optional, Union
+import json
+from collections import defaultdict
+import logging
+import time
+
+from tamr_unify_client import Client
+from tamr_unify_client.operation import Operation
+
+from tamr_toolbox.models.data_type import JsonDict
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def update_llm_data(
+    tamr_client: Client,
+    *,
+    project_name: str,
+    do_update_clusters: bool = True,
+    do_use_manual_clustering: bool = False,
+) -> None:
+    """
+    Updates data for LLM query if needed, based on latest published clusters.
+
+    Args:
+        tamr_client: Tamr client object
+        project_name: name of the project to be updated
+        do_update_clusters: whether to update clusters, default True
+        do_use_manual_clustering: whether to use externally managed clustering, default False
+    """
+
+    url = (
+        f"projects/{project_name}:updateLLM?updateClusters={do_update_clusters}"
+        f"&useManualClustering={do_use_manual_clustering}"
+    )
+    response = tamr_client.post(url)
+    if not response.ok:
+        message = (
+            f"LLM update for {project_name} failed at submission time: "
+            + response.json()["message"]
+        )
+        LOGGER.error(message)
+        raise RuntimeError(message)
+    operation_id = response.content.decode("latin1")
+
+    # An operation id of '-1' is returned when LLM is already up to date
+    if operation_id != "-1":
+        operation = Operation.from_resource_id(tamr_client, operation_id)
+        operation.wait()
+    return None
+
+
+def poll_llm_status(
+    match_client: Client, *, project_name: str, num_tries: int = 10, wait_sec: int = 1
+) -> bool:
+    """
+    Check if LLM is queryable. Try up to num_tries times at 1s (or user-specified) interval.
+
+    Args:
+        match_client: a Tamr client set to use the port of the Match API
+        project_name: name of target mastering project
+        num_tries: max number of times to poll endpoint, default 10
+        wait_sec: number of seconds to wait between tries, default 1
+    Returns:
```

Comment:
Docstring formatting:
```suggestion
        wait_sec: number of seconds to wait between tries, default 1
    
    Returns:
```